### ISSUE TITLE
stack expects each tensor to be equal size, but got [3, 32, 32] at entry 0 and [1, 32, 32] at entry 15

### ISSUE THREAD

**Issue by GolvenDong** [lucidrains/denoising-diffusion-pytorch#97]
When I train with my own dataset which size is 32*32,this problem occured.Can anyone help me to deal with this problem?

Here is the training script:

model = Unet(
    dim = 16,
    dim_mults = (1, 2, 4)
).cuda()

diffusion = GaussianDiffusion(
    model,
    image_size = 32,
    timesteps = 1000,           # number of steps
    sampling_timesteps = 250,   # number of sampling timesteps (using ddim for faster inference [see citation for ddim paper])
    loss_type = 'l1'            # L1 or L2
).cuda()

trainer = Trainer(
    diffusion,
    '/home/golven/dataset/crop_train',
    train_batch_size = 128,
    train_lr = 8e-5,
    train_num_steps = 700000,         # total training steps
    gradient_accumulate_every = 2,    # gradient accumulation steps
    ema_decay = 0.995,                # exponential moving average decay
    amp = False                        # turn on mixed precision
)

trainer.train()

**Comment by gau-nernst** [comment]
Your dataset probably consists of both RGB images (3 channels) and grayscale images (1 channel). A quick hack for me was to go to the line below and replace it with `img = Image.open(path).convert("RGB")`

https://github.com/lucidrains/denoising-diffusion-pytorch/blob/98f2eeac351c494542c97ffad3b0adf8dfc7cbe4/denoising_diffusion_pytorch/denoising_diffusion_pytorch.py#L720-L723
